doc/xgboost_on_sqlflow_design.md

```diff
@@ -1,152 +1,109 @@
-
-    
-class ModelSource:
-    """
-    ModelSource API
-    A handler by which XGBLauncher save/load model(booster) and related information.
-    """
-    def __init__(self, source_conf):
-        pass
-
-    @abstractmethod
-    def read_buffer(self, model_path: str) -> bytes:
-        pass
-
-    @abstractmethod
-    def write_buffer(self, buf: bytes, model_path: str):
-        pass
-
-    @abstractmethod
-    def read_lines(self, model_path: str) -> List[str]:
-        pass
-
-    @abstractmethod
-    def write_lines(self, lines: List[str], model_path: str):
-        pass
-``` 
-
-
-With the help of xgblauncher, we can launch xgboost from sqlflow AST via a lightweight python `code template` and a corrsponding `filler`.
-The `code template` roughly includes components as follows: 
-
-* `TFDataSource` that is responsible for fetching and pre-processing data via tf.feature_columns API.
-   Data will be fetched in mini-batch style by executing TF compute graph with mini-batch data feed by `sqlflow_submitter.db.db_generator`.
-
-* `DBDataSource` that is responsible for writing prediction results into specific data base.
-   The writing action can be implemented via `sqlflow_submitter.db.insert_values`.
-
-* `LocalModelSource` that is responsible for reading/writing _gboost models on local file system.
-
-* Configure template building and entry point of xgblauncher.
-
-
-#### Running distributed xgboost job on k8s cluster
-
-Distributed training is supported in xgboost via [rabit](https://github.com/dmlc/rabit), a reliable allreduce and broadcast interface for distributed machine learning.
-To run a distributed xgboost job with `rabit`, all we need to do is setup a distributed environment.  
-
-For now, xgboost has been bind to some popular distributed computing frameworks, such as Apache Spark, Apache Flink, Dask.
-However, specific computing frameworks are not always available in production environments. 
-So, we propose a cloud-native approach: running xgboost directly on `k8s cluster`. 
- 
-As `xgblauncher` is scalable and docker-friendly, xgblauncher-based containers can be easily orchestrated by [xgboost operator](https://github.com/kubeflow/xgboost-operator),
-a specific kubernetes controller for (distributed) xgboost jobs.
-With the help of `xgboost operator`, it is easy to handle `XGBoostJob` via `kuberentes API`, a kubernetes' custom resource defined by `xgboost operator`. 
-
-`XGBoostJob` building and tracking will be integrated to `xgblauncher` in near future. 
-After that, we can generate python codes with an option to decide whether running xgboost job locally or submitting it to remote k8s cluster.
+# Design Doc: XGBoost on SQLFlow
+
+## Introduction
+
+This design doc introduces how do users train/predict the [XGBoost](https://xgboost.ai/) model by SQLFlow SQL and how
+we implement it.
+
+## Design
+
+We prefer users to execute the SQLFlow Train/Predict SQL as follows:
+
+  ``` sql
+  SELECT * FROM train_table
+  TRAIN XGBoost
```

Comment:
`TRAIN XGBoost.someModel`?

Reply:
XGBoost use the `objective` paramter to specify the training objective such as:
`objective=binary:logistic `, ref https://xgboost.readthedocs.io/en/latest/parameter.html#learning-task-parameters

Maybe we can use `Train XGBoost` in the train statement, and specify the objective in WITH statement: `model.objective=binary:logistic` ?

Reply:
I still prefer to put the `objective` in the `TRAIN` clause, it seems quite similar to `tf.estimator.*`.

Reply:
Update the doc, would use `TRAIN xgboost.multi.softmax` to fill the objective parameter.

Reply:
> I still prefer to put the `objective` in the `TRAIN` clause, it seems quite similar to `tf.estimator.*`.

@Yancey1989 @typhoonzero `objective` corresponds to the loss of function of a model. So it shouldn't be in the model name. For different types of models, there are `gbtree`, `gblinear` and `dart` as listed [here](https://xgboost.readthedocs.io/en/latest/parameter.html#general-parameters).

Reply:
@tonyyang-svail Thanks, you are right. I'll update to put`objective` in attributes.